crates/sargon/src/system/sargon_os/profile_state_holder.rs

```diff
@@ -143,31 +143,38 @@ impl ProfileStateHolder {
     /// Sets the `ProfileState` held by this `ProfileStateHolder` to the latest `profile_state`.
     pub(super) fn replace_profile_state_with(
         &self,
+        is_android: bool,
         profile_state: ProfileState,
     ) -> Result<()> {
         let mut lock = self.state.write().expect(
             "Stop execution due to the profile state lock being poisoned",
         );
-        Self::assert_factor_instances_valid(&profile_state)?;
```

Comment:
We used to ASSERT on profile replaced, now instead, we "diagnose" which uses `log` to log - and in the future we can add runtime warnings if we pass such a new driver from hosts. (on iOS we can use https://github.com/pointfreeco/swift-issue-reporting to halt execution when running `DEBUG` from Xcode)